realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -268,6 +267,12 @@ void notifyAllListeners() {
         if (!realm.isClosed() && threadContainsAsyncEmptyRealmObject()) {
             updateAsyncEmptyRealmObject();
         }
+        // It is very important to notify the global listeners last.
+        // We don't sync RealmResults in realmChanged, instead, they are synced in notifySyncRealmResultsCallbacks.
+        // This is because of we need to compare the table view version to decide if it changes. Thus, we cannot sync
```

Comment:
This is because we need to compare the TableView version in order to decide if it changed.